### PR TITLE
Fixes for various issues

### PR DESCRIPTION
**Pull request by elcritch**
- Add logging for GitHub searches
- Add fallback search for non-language specific GH search with manual filtering
- Format requires using package name or url depending on if it's in packages.json
- Adding requires to the Nimble file adds after the last existing requires
- readme tweaks
- for tagged releases don't build fake arm binaries

### CODE REVIEW

**Comment by elcritch** [comment]
I also created a https://github.com/elcritch/exampleNimProject but am too lazy to add it to the unit tests.

File: src/packagesjson.nim [hunk 1]
```diff
@@ -1,5 +1,6 @@
 
 import std / [json, os, sets, strutils, httpclient, uri, options]
+import ./context
```
Comment (Araq) [inline src/packagesjson.nim:3]:
```suggestion
import context
```

We don't use Status's "delusions of persecution" style.

File: src/packagesjson.nim [hunk 2]
```diff
@@ -94,17 +95,43 @@ proc singleGithubSearch(term: string): JsonNode =
...
+        var filtered = newJArray()
+        for item in result.items():
+          let queryUrl = item["languages_url"].getStr
+          let langs = client.getContent(queryUrl).parseJson()
+          if langs.hasKey("Nim"):
+            filtered.add item
+        result = filtered
+      
+      if result.len() == 0:
+        if fullSearch == false:
```
Comment (Araq) [inline src/packagesjson.nim:118]:
```suggestion
        if not fullSearch:
```

File: src/packagesjson.nim (hunk 1 above)
Comment (elcritch) [reply src/packagesjson.nim]: I didn’t realize it had a name 🤣

**Pull request merged**